etcdserver/auth/auth.go

```diff
@@ -195,6 +185,10 @@ func (s *Store) CreateUser(user User) (User, error) {
 	if err == nil {
 		plog.Noticef("created user %s", user.User)
 	}
+	// Require that root always has a root role.
```

Comment:
i do not quite understand this comment. 
I would suggest you to move 189 to 184 and change comment to // Attach root role to root user